### ISSUE TITLE
ValueError: math domain error, when setting theil_u=True

### ISSUE THREAD

**Issue by rawanm** [shakedzy/dython#12]
Hi Shaked, 
Thank you for providing a great library and informative write-up! 

I faced an issue with associations method, check below for details:

Usage: 
* All selected features are categorical of type (Object) 
```
from dython.nominal import associations
accidents_data_catigorical = ['Speed_limit', 'Junction_Detail', 'Light_Conditions', 
                              'Weather_Conditions', 'Road_Surface_Conditions', 'Special_Conditions_at_Site',
                             'Carriageway_Hazards', 'Urban_or_Rural_Area', 'Accident_Severity', '1st_Road_Class']

corr = associations(accidents_data[accidents_data_catigorical], theil_u=True, 
                    return_results=True, plot=True, nominal_columns=accidents_data_catigorical)

```

Thrown error trace: 
---------------------------------------------------------------------------
```
ValueError                                Traceback (most recent call last)
<ipython-input-105-99541e13a9e9> in <module>()
      5 
      6 corr = associations(accidents_data[accidents_data_catigorical], theil_u=True, 
----> 7                     return_results=True, plot=True, nominal_columns=accidents_data_catigorical)

/Applications/anaconda3/lib/python3.6/site-packages/dython/nominal.py in associations(dataset, nominal_columns, mark_columns, theil_u, plot, return_results, **kwargs)
    169                         if theil_u:
    170                             corr[columns[j]][columns[i]] = theils_u(dataset[columns[i]],dataset[columns[j]])
--> 171                             corr[columns[i]][columns[j]] = theils_u(dataset[columns[j]],dataset[columns[i]])
    172                         else:
    173                             cell = cramers_v(dataset[columns[i]],dataset[columns[j]])

/Applications/anaconda3/lib/python3.6/site-packages/dython/nominal.py in theils_u(x, y)
     76         in the range of [0,1]
     77     """
---> 78     s_xy = conditional_entropy(x,y)
     79     x_counter = Counter(x)
     80     total_occurrences = sum(x_counter.values())

/Applications/anaconda3/lib/python3.6/site-packages/dython/nominal.py in conditional_entropy(x, y)
     29         p_xy = xy_counter[xy] / total_occurrences
     30         p_y = y_counter[xy[1]] / total_occurrences
---> 31         entropy += p_xy * math.log(p_y/p_xy)
     32     return entropy
     33 

ValueError: math domain error

```

Returning to the default value for theil_u=false, no error is thrown, however, since correlation is between categorical features it's beneficial to use Uncertainty Coefficient (Theil’s U)

**Comment by shakedzy** [comment]
Hi @rawanm - could you please supply the `accidents_data` dataframe? preferably as a csv file I can load directly to pandas. 
Also, as lesson-learned from another issue - please make sure you use Python 3.x. Python 2.7 is not supported and might cause these unexplained errors.

**Comment by rawanm** [comment]
Python version: 3.6.8
Since my dataset was too large I subsetted a small for your reference, but it didn't yield any error. So I did some digging to identify which part of the dataset that was causing an issue, and silly me I found there were still some NaN values in selected features  (I decided to leave some imputing/cleaning for later and seemed to have forgotten)! Anyhow, I'm currently able to use (theil_u=True) without any issues.

Thanks again @shakedzy, appreciate it!
